File: go/subscription_coupon.go
// this file is @generated
package meteroid

import "time"

// Coupon as embedded in subscription details — a subset of the `Coupon` resource
// returned by the coupons API.
type SubscriptionCoupon struct {
	Code string `json:"code"`

	Description string `json:"description"`

	Disabled bool `json:"disabled"`

	Discount CouponDiscount `json:"discount"`

	// RFC 3339 timestamp.
	ExpiresAt *time.Time `json:"expires_at,omitempty"`

	Id CouponId `json:"id"`

	RecurringValue *int32 `json:"recurring_value,omitempty"`

	RedemptionLimit *int32 `json:"redemption_limit,omitempty"`

	Reusable bool `json:"reusable"`
}

File: python/meteroid/models/subscription_coupon.py
# this file is @generated
import dataclasses
import typing as t
from datetime import datetime

from ..serialization import BaseModel
from .coupon_discount import CouponDiscount
from .coupon_id import CouponId


@dataclasses.dataclass
class SubscriptionCoupon(BaseModel):
    """Coupon as embedded in subscription details — a subset of the `Coupon` resource
    returned by the coupons API."""

    code: str

    description: str

    disabled: bool

    discount: CouponDiscount

    id: CouponId

    reusable: bool

    expires_at: t.Optional[datetime] = None

    recurring_value: t.Optional[int] = None

    redemption_limit: t.Optional[int] = None
